### ISSUE TITLE
unawaited_futures detected when calling then() in an async body

### ISSUE THREAD

**Issue by jfacoustic** [dart-lang/linter#4261]
Not sure if this is a bug or the expected behavior for the lint rule, but when I call `future.then` in an async body, it gives the unawaited_futures warning

```dart
  Future<int> getInt() async {
    return 3;
  }

  Future<void> foo() async {
    // warning shown here.
    getInt().then((v) => print(v));
  }
```

I would have anticipated then() to automatically infer the future is meant to be unawaited.

**Comment by lrhn** [comment]
This is working as intended. The `.then` call returns a new future, which is unawaited.
You only get the warning inside an `async` function, where you are not expected to use `then`.

**Comment by pq** [comment]
I think @lrhn is right here. @jfacoustic, feel free to re-open if you feel otherwise!
